src/imageuploadengine.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) 2016, CKSource - Frederico Knabben. All rights reserved.
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import { eventNameToConsumableType } from '@ckeditor/ckeditor5-engine/src/conversion/model-to-view-converters';
+import FileRepository from './filerepository';
+import ImageUploadCommand from './imageuploadcommand';
+import Notification from '@ckeditor/ckeditor5-ui/src/notification/notification';
+import uploadingPlaceholder from '../theme/icons/image_placeholder.svg';
+import { isImageType } from './utils';
+
+/**
+ * Image upload engine plugin.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class ImageUploadEngine extends Plugin {
+	constructor( editor ) {
+		super( editor );
+
+		/**
+		 * Image's placeholder that is displayed before real image data can be accessed.
+		 *
+		 * @member {String} #placeholder
+		 */
+		this.placeholder = 'data:image/svg+xml;utf8,' + uploadingPlaceholder;
```

Comment:
It can we `protected` at this stage. We can make it public when we will find it needed.